umap/static/umap/js/modules/rules.js

```diff
@@ -0,0 +1,240 @@
+import { DomUtil, DomEvent, stamp } from '../../vendors/leaflet/leaflet-src.esm.js'
+import * as Utils from './utils.js'
+import { translate } from './i18n.js'
+
+class Rule {
+  constructor(map, condition = '', options = {}) {
+    this.map = map
+    this.active = true
+    this.options = options
+    let isDirty = false
+    Object.defineProperty(this, 'isDirty', {
+      get: () => {
+        return isDirty
+      },
+      set: (status) => {
+        isDirty = status
+        if (status) {
+          this.map.isDirty = status
+        }
+      },
+    })
+    let _condition
+    Object.defineProperty(this, 'condition', {
```

Comment:
Rather than using `Object.defineProperty` you could directly use a [get property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get), which could make the code more readable.

Any reason not to?

Reply:
Excellent! I've use it :)